src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -156,28 +156,16 @@ private async Task<ICredential> GetRefreshedCredentials(Uri remoteUri, string us
 
                     switch (result.AuthenticationMode)
                     {
+                        case AuthenticationModes.Basic:
```

Comment:
So what used to happen was:

The user would first be shown a Basic Auth dialog in the helper.
They would enter username/password which are returned from the dialog.

If they were using a URL that isn't bitbucket.org  the username/password is not tested, simply passed back to the GCM app and then back to Git.

If they were using bitbucket.org host, then the username/password is tested by calling the REST API.
If it returns a 200 then the username/password is passed back to the GCM app and then back to Git.
If it returns a 403 it would show the OAuth dialog in the helper.
Anything else would throw an exception.


Removing this code, means the behaviour for Bitbucket.org is the same as for other URLs.

Users will now need to manually choose the OAuth option on the Basic Auth dialog, or force OAuth via credential.bitbucketAuthModes.



I was planning to provide a follow up PR to switch the default behaviour of the Helper dialog and make OAuth the first option, Users would then need to manually choose Basic Auth.

Reply:
The problem here is that the user would not necessarily know the reason their username/password didn't work (because they need to auth via OAuth).

Might be worth only presenting the OAuth option for Cloud and only U/P for Server and DC.

Reply:
App Passwords will still work on Bitbucket.org with Basic Auth

Reply:
We could also provide a link to the documentation referring to the removal of rights from the user's main u/p ?

Reply:
> But I do think when the host is bitbucket.org the UI should be reversed to put OAuth first with the option to switch to Basic Auth and renaming "Basic Auth" to "App Password".

I think this is the right path forward - is this something you'd be able to take on?

Reply:
>  is this something you'd be able to take on?

I can do. I can't make guarantees about deliver timescales unfortunately.

Do you want it addressed in this PR? I wouldn't really want to delay this fix too long.

Reply:
No, I think we can take this fix. Thank you, Mike!